client/src/app/components/task-list/task-list/task-list.component.ts

```diff
@@ -1,26 +1,75 @@
 import { Component, OnInit } from '@angular/core';
 import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
-import { ITask } from 'src/app/interfaces/task';
+import { TaskDTO } from 'src/app/interfaces/task';
 import { TaskService } from 'src/app/services/task.service';
 import { TaskModalComponent } from '../../task-modal/task-modal/task-modal.component';
+import Swal from 'sweetalert2';
 
 @Component({
   selector: 'app-task-list',
   templateUrl: './task-list.component.html',
   styleUrls: ['./task-list.component.scss'],
   providers: [TaskService],
 })
-export class TaskListComponent {
-  tasks: ITask[] | undefined = [];
+export class TaskListComponent implements OnInit {
+  tasks: TaskDTO[] | undefined = [];
   constructor(private taskService: TaskService, private ngbModal: NgbModal) {}
 
+  async ngOnInit(): Promise<void> {
+    const tasks = (await this.taskService.getTasks()).toPromise();
+    tasks.then((tasks) => {
+      this.tasks = tasks;
+    });
+  }
+
+  deleteTask(id: number) {
+    Swal.fire({
+      title: 'Are you sure?',
+      text: "You won't be able to revert this!",
+      icon: 'warning',
+      showCancelButton: true,
+      confirmButtonColor: '#f44336',
+      cancelButtonColor: '#ff9800',
+      confirmButtonText: 'Yes, delete it!',
+    }).then(async (result) => {
+      if (result.isConfirmed) {
+        (await this.taskService.deleteTask(id)).toPromise().then(() => {
+          Swal.fire({
+            title: 'Deleted!',
+            text: 'Item has been deleted.',
+            icon: 'success',
+          }).then((value) => {
+            if (value.isConfirmed) {
+              location.reload();
+            }
+          });
+        });
+      }
+    });
+  }
+
+  editTask(task: TaskDTO) {
+    const modalRef = this.ngbModal.open(TaskModalComponent, {
+      backdrop: 'static',
+      keyboard: false,
+    });
+    modalRef.componentInstance.task = task;
+    modalRef.result.then((result) => {
+      if (result) {
+        location.reload();
+      }
+    });
+  }
+
   createTask() {
     const modalRef = this.ngbModal.open(TaskModalComponent, {
       backdrop: 'static',
       keyboard: false,
     });
     modalRef.result.then((result) => {
-      console.log(result);
+      if (result) {
+        location.reload();
```

Comment:
temporary solution

Reply:
For temporary solutions, I would recommend making sure you can't accidentally forget to change it later. A code comment would be a good simple solution (otherwise people reading your code in the future have no idea you intend to change it, you can put `// TODO: make API call instead of reloading`), for important TODOs you can add it to the task tracker.